### PR TITLE
Refactored away circular dependencies in Multiplayer gem cmake targets.

### LINKED ISSUE #5611: Building any project using the Project Manager built with target INSTALL fails
**Describe the bug**
It is not possible to build any project via the Project Manager when it is launched from the o3de-install folder (after following the [Pre-built SDK engine guide](https://o3de.org/docs/welcome-guide/setup/setup-from-github/#:~:text=Use%20CMake%20to%20build,target%20INSTALL%20--config%20profile%20--%20/m)) due to the following error:
```
CMake Error at C:/o3de-install/cmake/LYWrappers.cmake:690 (if):
  Maximum recursion depth of 1000 exceeded
Call Stack (most recent call first):
  C:/o3de-install/cmake/SettingsRegistry.cmake:66 (ly_de_alias_target)
  C:/o3de-install/cmake/SettingsRegistry.cmake:72 (ly_get_gem_load_dependencies)
```
Please note that the Project Manager built following the instructions for building the [Source Engine](https://o3de.org/docs/welcome-guide/setup/setup-from-github/#:~:text=The%20following%20example,Editor%20--config%20profile%20--%20/m) does build projects successfully without the mentioned issue.

Please refer to the attached video and logs for more information.

**Steps to reproduce**
1. Follow the [Pre-built SDK engine guide](https://o3de.org/docs/welcome-guide/setup/setup-from-github/#:~:text=Use%20CMake%20to%20build,target%20INSTALL%20--config%20profile%20--%20/m).
2. Launch O3DE.exe from the o3de-install/bin/Windows/profile/Default folder.
3. Create a new project.
4. Build the created project.

**Expected behavior**
The project is built without any issues.

**Actual behavior**
The project is not built due to the errors.

**Video**

https://user-images.githubusercontent.com/86953659/141764735-10954811-9ef3-44cf-954a-f7b33a2fc2ce.mp4

**Found in Branch**
Stabilization/2110 (76b7e73)

**Desktop**
Device: PC
OS: Windows
Version: 10
CPU: Intel i5-9600K
GPU: RTX 2070
Memory: 16GB

**Additional context**
[CMakeProjectBuildError.log](https://github.com/o3de/o3de/files/7537837/CMakeProjectBuildError.log)
[CMakeOutput.log](https://github.com/o3de/o3de/files/7537839/CMakeOutput.log)

### PR DESCRIPTION
**Pull request by AMZN-Olex**
+ Tested with clean empty project Cache folder
+ Resolves #5611

Signed-off-by: Olex Lozitskiy <5432499+AMZN-Olex@users.noreply.github.com>

### CODE REVIEW

File: Gems/Multiplayer/Code/CMakeLists.txt [hunk 1]
```diff
@@ -154,7 +114,8 @@ if (PAL_TRAIT_BUILD_HOST_TOOLS)
                 Gem::Multiplayer.Static
                 Gem::Multiplayer.Builders
     )
-
+    
+    ly_create_alias(NAME Multiplayer.Builders  NAMESPACE Gem TARGETS Gem::Multiplayer.Editor)
     # use the Multiplayer.Editor module in tools like the Editor:  Such tools also get the visual debug view:
     ly_create_alias(NAME Multiplayer.Tools     NAMESPACE Gem TARGETS Gem::Multiplayer.Editor Gem::Multiplayer.Debug Gem::Multiplayer.Builders)
```
Comment (lumberyard-employee-dm) [inline Gems/Multiplayer/Code/CMakeLists.txt:120]: Now that the `Multiplayer.Builders` variant is aliasing the `Gem::Multiplayer.Editor` target, the `Multiplayer.Tools` variant can remove `Gem::Multiplayer.Editor` target from its list of targets

Comment (AMZN-Olex) [reply Gems/Multiplayer/Code/CMakeLists.txt]: Done.

**Review by AMZN-Gene — APPROVED** [review]

File: Gems/Multiplayer/Code/CMakeLists.txt [hunk 2]
```diff
@@ -197,6 +158,7 @@ if (PAL_TRAIT_BUILD_TESTS_SUPPORTED)
             NAMESPACE Gem
             FILES_CMAKE
                 multiplayer_tools_tests_files.cmake
+                multiplayer_tools_files.cmake
```
Comment (amznestebanpapp) [inline Gems/Multiplayer/Code/CMakeLists.txt:161]: these files are being compiled by Multiplayer.Editor and by this target, which means that they are being compiled twice. Instead, create a static library that builds multiplayer_tools_files and then link against it in the Editor module and this test

**Review by amznestebanpapp — CHANGES_REQUESTED** [review]

File: Gems/Multiplayer/Code/CMakeLists.txt (hunk 2 above)
Comment (amznestebanpapp) [reply Gems/Multiplayer/Code/CMakeLists.txt:161]: took a second look and the number of files there is small so I will drop it, but this should still be done eventually to de-duplicate and reduce compilation time.

**Review by amznestebanpapp — APPROVED** [review]

**Review by SergeyAMZN — APPROVED** [review]

**Review by lumberyard-employee-dm — APPROVED** [review]

**Review by amznestebanpapp — APPROVED** [review]
thanks!

**Review by lumberyard-employee-dm — APPROVED** [review]

File: Gems/Multiplayer/Code/CMakeLists.txt (hunk 2 above)
Comment (AMZN-Olex) [reply Gems/Multiplayer/Code/CMakeLists.txt]: Done.

**Pull request merged**